TypeCobol/Compiler/Parser/CodeElementsParserStep.cs

```diff
@@ -337,6 +338,11 @@ internal static IList<DocumentChange<ICodeElementsLine>> ParseProcessedTokensLin
                                     MessageCode.TypeCobolParserLimitation);
                             }
 
+                            if (processedTokensDocument.TextSourceInfo.ColumnsLayout == ColumnsLayout.CobolReferenceFormat && codeElement.StartingArea != TextAreaType.Source)
```

Comment:
The test on starting area different from Source should be handled by the `CheckAreaOfDeclaration` (done in my proposed version)